test/Microsoft.ML.Predictor.Tests/Microsoft.ML.Predictor.Tests.csproj

```diff
@@ -11,6 +11,7 @@
     <ProjectReference Include="..\..\src\Microsoft.ML.LightGBM\Microsoft.ML.LightGBM.csproj" />
     <ProjectReference Include="..\..\src\Microsoft.ML.PCA\Microsoft.ML.PCA.csproj" />
     <ProjectReference Include="..\..\src\Microsoft.ML.StandardLearners\Microsoft.ML.StandardLearners.csproj" />
+    <ProjectReference Include="..\..\src\Microsoft.ML.StaticPipe\Microsoft.ML.StaticPipe.csproj" />
```

Comment:
>StaticPipe [](start = 78, length = 10)

This is not a good sign.
In current moment we prefer to separate our StaticPipe project from rest of the code.

I would suggest to split LocalPathReader into two classes. One should stay next to TextLoader.cs and have just Read method without <Shape>.
Second should remain in StaticPipe project, and have Read method with <Shape>.

Reply:
Just to be clear on your meaning, when you say with/without "." you're talking about calling something like `reader.Read(...)` vs just `Read(...)`?

Reply:
Sorry for confusion, I forgot what web interface doesn't gracefully handles tags, and you need to escape them all the time.
I was talking about `public static DataView<TShape> Read<TShape>` that thing should remain in StaticPipe project.

If I correctly understand you code and intention you removing `public IDataView Read(string path)` from TextLoader but since `LocalPathReader` contains extension method on `IDataReader<IMultiStreamSource> reader` all our code continue to work since we fall back to that extension.

This is great, it just that extension shouldn't be part of StaticPipe project. I would prefer it to remain in Microsoft.ML.Data project.
In same time second extension method which works on top of `DataReader<IMultiStreamSource, TShape>` should remain to be inside StaticPipe project.

Does my explanation make sense?